test/e2e/auth/e2edata/per_node_validatingadmissionpolicy.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "only-allow-name-matching-node-configmaps"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        # notice you cannot restrict DELETEs.  Your per-node actor should not have deletion powers.
```

Comment:
why is this comment here? deletes can be restricted...

Reply:
I can confirm that DELETEs get restricted exactly like CREATEs and UPDATEs - works for me.

Reply:
Hmm, I spoke to soon. I hadn't updated by RBAC properly when running my test. Now that RBAC allows DELETEs, I am getting an error when evaluating the CEL expression:
```
message: 'resourceslices.resource.k8s.io "dra-fictional-node-slice" is forbidden:
                ValidatingAdmissionPolicy ''resourceslices-policy-dra-kubelet-plugin-dra-6661.k8s.io''
                with binding ''resourceslices-policy-dra-kubelet-plugin-dra-6661.k8s.io'' denied
                request: expression ''!variables.isKubeletPlugin || !variables.hasNodeName || has(object.spec.nodeName)
                && object.spec.nodeName == request.userInfo.extra["authentication.kubernetes.io/node-name"][0]''
                resulted in error: no such key: spec'
```

So it looks like VAP really cannot work for DELETEs because it doesn't make the object that is being deleted available in the CEL environment.

Perhaps it should, but currently it doesn't?

Reply:
object is not set in DELETE operations, oldObject is

Reply:
I'd suggest adding DELETE here and dropping the comment, then using the request.name attribute instead of pulling data out of the object, like this:

```
  matchConditions:
    - name: isRestrictedUser
      expression: request.userInfo.username == 'system:serviceaccount:e2e-ns:default'
  variables:
    - name: userNodeName
      expression: request.userInfo.extra[?'authentication.kubernetes.io/node-name'][0].orValue('')
  validations:
    - expression: variables.userNodeName == ''
      message: "no node association found for user, this user must run in a pod on a node"
    - expression: variables.userNodeName != request.name
      message: "this user may only modify configmaps that belong to the node the pod is running on"
```

Reply:
> I'd suggest adding DELETE here and dropping the comment, then using the request.name attribute instead of pulling data out of the object.

Clever, though perhaps I'll open a followup showing how to use a field. The field is useful in cases where many instances of a resource can share a single node.

Reply:
Sure, you can pull a field value out for delete as well. Something like this hoists it into a variable later validations can reference succinctly with nice safe fallbacks to empty string:

```yaml
  variables:
  ...
    - name: objectNodeName
      expression: "request.operation == 'DELETE' ? oldObject : object).?whatever.nodeName.fieldPath.orValue('')"
```